### ISSUE TITLE
LLT_ALIGN macro in <julia.h> does not compile with MSVC14

### ISSUE THREAD

**Issue by phlavenk** [JuliaLang/julia#19217]
Hi,
On MSVC 2015-Update3 ( 14.0.25431.01 Update 3), Julia 0.5.0 binary Win64 install, include of <julia.h> in my own code ends with compilation error:
```
Error	C4146	unary minus operator applied to unsigned type, result still unsigned	EJBCommJuliaInterface	c:\users\phlavenk\documents\programs\julia-0.5\include\julia\julia.h	788	
```
related to 
```
julia.h, 786:
STATIC_INLINE char *jl_symbol_name_(jl_sym_t *s)
{
	return (char*)s + LLT_ALIGN(sizeof(jl_sym_t), sizeof(void*));
	//return (char*)s + (((sizeof(jl_sym_t)) + (sizeof(void*)) - 1) & -(sizeof(void*)));// (((x)+(sz)-1) & -(sz))
}
```
the `LLT_ALIGN` macro with definition `#define LLT_ALIGN(x, sz) (((x) + (sz)-1) & -(sz))` has a problem in the `-(sz)` part. 
Happens during my attempt to use CxxWrap.jl, where CxxWrap\usr\include\type_conversion.hpp includes julia.h

I'm not sure what is the action this macro tries to achieve and thus I'd like ask for advice how to bugfix this makro.
Petr

**Comment by yuyichao** [comment]
According to [cppreference](http://en.cppreference.com/w/cpp/language/operator_arithmetic) unary minus is well defined for unsigned numbers as

> For unsigned `a`, the value of `-a` is `2^b - a`, where `b` is the number of bits after promotion.

So MSVC is wrong to report it as an error. What other compiler options do you have? This macro has been there for a long time and I don't remember @tkelman complained about it.

We could replace it with sth like `(0 - x)` or `~(x - 1)` (if I didn't have any off-by-one or off-by-a-sign error)....

**Comment by tkelman** [comment]
Looks like it's typically a warning (https://msdn.microsoft.com/en-us/library/4kh09110.aspx), and a bit of a silly one if the wraparound unsigned minus behavior was intended. You probably have your warning level higher than I usually test with MSVC, and the equivalent of `-Werror` enabled?

**Comment by mindbound** [comment]
Still having this as an error with Julia 0.6.2, MSVC15, even with `/WX-` (equivalent to not having `-Werror` enabled). To avoid it, I needed to also disable the Security Development Lifecycle recommended checks, with `/sdl-`.

**Comment by taoyouh** [comment]
This should be already fixed by PR  #33511
